aips/aip-hybrid-nft-standard.md

```diff
@@ -0,0 +1,352 @@
+
+```move
+#[view]
+public fun is_hybrid_asset<T: key>(collection: Object<T>): bool
+
+#[view]
+public fun is_hybrid_token<T: key>(token: Object<T>): bool
+```
+These functions enable verification of hybrid asset status and querying of token states, helping interfaces and external contracts interact with hybrid assets correctly.
+
+```move
+#[view]
+public fun get_nfts_by_owner(
+    owner_address: address, 
+    collection: Object<HybridCollection>
+): vector<address>
+
+#[view]
+public fun get_treasury_balance(collection: Object<HybridCollection>): u64
+```
+These functions provide essential information for:
+- NFT ownership tracking
+- Treasury management
+- Balance verification
+- Collection status monitoring
+
+### Capability Management
+
+Capability management functions control the minting and burning permissions of the collection.
+
+```move
+public entry fun destroy_mint_capability(
+    caller: &signer,
+    collection: Object<HybridCollection>
+)
+
+public entry fun destroy_burn_capability(
+    caller: &signer,
+    collection: Object<HybridCollection>
+)
+```
+These functions provide critical security controls by allowing:
+- Permanent removal of minting capability
+- Restriction of burning operations
+- One-way capability destruction for enhanced security
+
+## Reference Implementation
+
+GitHub Repository: https://github.com/gregnazario/aptos-hybrid-assets
+
+## Risks and Drawbacks
+
+- Potential drawback is adoption of marketplaces and wallets to use the new transfer function. With dynamic dispatch it
+  could be replaced with a more flexible solution.
+    - The mitigation plan is to provide help to wallets and marketplaces to adopt the new standard via a helper
+      function.
+- For future backward compatibility, depending on how the dynamic dispatch solution is provided, it may be necessary to
+  change the contract, or create a new contract altogether.
+    - The mitigation plan is to in the future provide a migration plan for future hybrid assets.
+
+## Security Considerations
+
+This AIP doesn't impact the security of the network, there are a few considerations for the project implemented this AIP,
+
+The implementation must specifically address:
+- Prevention of test and abort attacks on randomness mechanisms
+- Secure handling of Digital Assets metadata revelation
+- Atomic execution of paired asset transfers
+- Protection against balance manipulation attacks
+
+## Integration Considerations
+
+### DEX Integration
+- Must support dispatchable fungible assets
+- Optimize for high-volume trading scenarios
+- Consider gas efficiency in automated minting/burning operations
+- Optional NFT minting/burning bypass for DEX addresses
+
+### Digital Assets Marketplace Integration
+- Implement support for atomic transfers
+- Handle metadata revelation appropriately
+- Address the absence of dynamic dispatch in object transfers
+
+### Wallet Integration
+- Implement dual-asset display capabilities
+- Support reveal mechanics
```

Comment:
Should this be part of this proposal or a separate standard for revealing all nfts? I might be missing some nuances here but revealing doesn't seem new to the hybrid dn404 standard.